android/src/main/java/rnxmpp/service/XmppServiceSmackImpl.java

```
@@ -150,6 +150,11 @@ public XmlStringBuilder toXML() {
 
     @Override
     public void sendStanza(String stanza) {
+        if (connection == null) {
+            logger.log(Level.WARNING, "Connection has not been initialized yet, cannot send stanza " + stanza);
```

Wouldn't it be better to throw an Exception than logging and silently discarding the stanza? What if logging is set to level ERROR?